### DISCUSSION TITLE
Sacred Integration for Reproducibility

### DISCUSSION THREAD

**Discussion by vlomonaco** [ContinualAI/avalanche#404] [Feature Request]
At the moment Avalanche have a little support for experiments reproducibility. This can be done mostly by config files, logs and a [little support form the benchmarks](https://vlomonaco.github.io/avalanche/autoapi/avalanche/benchmarks/?highlight=reproducibility_data#avalanche.benchmarks.GenericCLScenario.get_reproducibility_data).

I was thinking it would nice to integrate those mechanisms with [Sacred](https://github.com/IDSIA/sacred). It is a great tool for reproducibility. What do you think?

**Comment by AntonioCarta** [comment]
I know it and I think it's useful but I never used it. I would not add anything new to the Evaluation/Metrics side for the alpha. First, we have to decide how to do it.

Also, is this something that needs to be done by a CL library? Personally, I would not use a library that forced me to a specific tool to track my own experiments because I already have my tools. Basically, I think avalanche should be a library and not a framework.

**Comment by AndreaCossu** [comment]
I agree with @AntonioCarta on this. What we could provide are utilities to launch experiment from configuration file (e.g. yaml) and complete logs and, when needed, plots. I don't think it is required to have all this stuff ready for the alpha but most definitely they will be needed in future releases.

**Comment by vlomonaco** [comment]
Yes, I agree with both of you! But when developing we should keep in mind that people may need to use it. So we need to carefully plan how to make the integration of the two tools as easy as possible.
